README.md

```diff
@@ -124,8 +124,11 @@ The spec file for the e2e needs to be a yaml file with the following structure:
   - `config` : The config values for this NR integration that will be red by the agent to execute the integration.
 - `tests` : The 3 kinds of tests that will be done to the New relic api to check for metrics/entities in NROne:
   - `nrqls` : Array of queries that will be executed independently. You can specify if running a query an error is expected or not.
-  - `query` : the query to run
-  - `error_expected`: false by default, useful if we want to test that a metric is not being sent
+    - `query` : the query to run
+    - `error_expected`: false by default, useful if we want to test that a metric is not being sent. This cannot be used in conjunction with `expected_results`.
+    - `expected_results` : Array of expected results that will be sequentially asserted against the NRQL response.
+      - `key`: The key of the expected result to assert against (i.e. `Pods Available`)
+      - `value`: The value expected for the above key (i.e. `4`)
```

Comment:
In the past we used to do it leveraging the nrql itself, could it be an option? https://github.com/newrelic/newrelic-prometheus-configurator/blob/db2606b31c7e55fbee9f2b572caf5d7875d04a16/test/e2e/test-specs.yml#L46C2-L46C2

```
        - query: "FROM Metric select metricName where collector_type = 'integration-filters' and metricName = 'timeseries_write_test' and getField(timeseries_write_test,'type') = 'cumulativeCount'"
          error_expected: false #In this case we expect the metrics thanks to the newrelic.io/scrape=true
```
Theoretically, it should be enough to see if a query with a `where` clause returns data instead of  introducing a new feature

Reply:
Hi Paolo, We've actually explored this possibility but decided against it for a couple of reasons: 1) Our use case requires us to test  FACET and other complex queries, thus, adding an additional WHERE clause obfuscates the NRQL SUT with the expected result. 2) We have nearly 200 NQL queries that will use this framework, and with such a large NRQL test, maintenance of a test suite that is this tightly coupled with the test logic becomes quite burdensome. As a heads up, our use case also requires us to test values that aren't exactly known so a future PR adds the ability to specify upper and lower bounds for certain NRQL responses (i.e CPU % utilization fluctuates but, should never be negative or above 100%). Happy to answer any questions on discuss this further!

Reply:
Hi @paologallinaharbur , Just bumping this for approval? Thanks!